encodings/protobuf/README.md

```diff
@@ -0,0 +1,178 @@
+# OCSF Protobuf Encoding
+
+This directory contains a sample Protobuf encoding for OCSF.
+
+Although encoding in general is _informative_ in OCSF, enabling the
+interoperation between systems in a single organization and indeed
+between organizations indicates that a **canonical** protobuf encoding
+has significant utility.  This directory contains that canonical
+encoding.
+
+## Semantic Type Considerations
+
+### Primitive Types
+
+The primitive types are mapped as follows
+
+| OCSF Type   | Protobuf Type           |
+| ----------- | ----------------------- |
+| `boolean_t` | `bool`                  |
+| `float_t`   | `float`                 |
+| `integer_t` | `int32`                 |
+| `long_t`    | `int64`                 |
+| `string_t`  | `string`                |
+| `json_t`    | `google.protobuf.Value` |
```

Comment:
I think it would be good to put a sentence on the choice of well-known Value for `json_t`.

Explicitly mentioning that `json_t` can be any json value (not just a document) will help people like me that weren't aware of this fact.

Reply:
Missing is the `object` object type, which is directly used in a couple cases.

The `object` type means "a key-value map where the values can be OCSF type: primitives, arrays, and objects". It's basically like `json_t` but must start with an object (a JSON object). I would guess this should use `google.protobuf.Value` as well.

Edit: I just noticed that the paragraph below mentions `object`. So perhaps add it to the table?

Edit 2: Should the table mention arrays?

Reply:
I will note that an array is not a type in OCSF, it is a type constructor.  That said, I should definitely call it out how arrays get encoded!